content/en/service_management/app_builder/expressions.md

```diff
@@ -0,0 +1,121 @@
+
+Some fields, like the [post-query transformation][1], display a code editor by default and accept plain JS. In all other fields, enclose your JS expressions in `${}`. For example, to interpolate the values of two text input components named `textInput0` and `textInput1` into the **Content** property of a text component (and add an exclamation mark), use the expression `${textInput0.value} ${textInput1.value}!`.
+
+{{< img src="service_management/app_builder/interpolation.png" alt="The text component fills with the words 'Hello' and 'World', each interpolated from a text input component value" style="width:100%;" >}}
+
+App Builder accepts standard vanilla JavaScript syntax, however:
+- The result of the expression must match the result expected by the component or query property. For example, the text component's **Is Visible** property expects a Boolean. To find out what type of data a component property expects, see [View component properties](#view-component-properties).
+- Your code has read-only access to the app state, but App Builder executes the code in a sandboxed environment with no access to the Document Object Model (DOM) or browser APIs.
+
+## View component properties
+
+Before you create an expression, it's helpful to know the available properties and defaults or current values for the component you want to interact with.
+
+You can view the available properties and values for a component using:
+- **App State**: Provides properties and values for all components and queries in your app, as well as global variables such as state variables or dashboard template variables.
+- **Inspect Data**: Provides properties and values for a specific component or query in your app.
+- The **Admin Console**: The **Data** tab of the **Admin Console** provides properties and values for all components and queries in your app.
+
+{{% collapse-content title="App State" level="h3" %}}
+To access **App State**:
+1. Click **App Properties** in the left side-panel.
+1. Scroll down to the **App State** section.
+
+{{< img src="service_management/app_builder/app-state.png" alt="The App State section in App Properties" style="width:100%;" >}}
+{{% /collapse-content %}}
+
+{{% collapse-content title="Inspect Data" level="h3" %}}
+To access **Inspect Data**:
+1. Click on the query or component you want to inspect.
+1. Scroll down to the **Inspect Data** section.
+
+{{< img src="service_management/app_builder/inspect-data.png" alt="The App State section in App Properties" style="width:100%;" >}}
+{{% /collapse-content %}}
+
+{{% collapse-content title="Admin Console" level="h3" %}}
+To access the **Admin Console**:
+1. Click on the cog (**Settings**) icon and select **Admin Console**.
+1. Click **Data**.
+
+{{< img src="service_management/app_builder/admin-console.png" alt="The App State section in App Properties" style="width:100%;" >}}
+{{% /collapse-content %}}
```

Comment:
IMO, these screenshots aren't adding much to the user experience, but potentially add a lot to maintenance overhead.

Reply:
Honestly, the UI right now can be a little unintuitive. I wanted to give at least a couple of examples of the kind of places you can enter expressions. Perhaps if the UI is improved later, the maintenance could be to remove the images.